signac/contrib/project.py

```diff
@@ -376,10 +382,12 @@ def id(self):
             The project id.
 
         """
-        try:
-            return str(self.config["project"])
-        except KeyError:
-            return None
+        if self._id is None:
+            try:
+                self._id = str(self.config["project"])
+            except KeyError:
```

Comment:
Do we need to raise an error here?

Reply:
No. The previous behavior was to return `None` if the `"project"` key is missing in the config. This change retains the previous behavior, because it `pass`es and returns `self._id`, which is already `None`. It may be more clear to make that explicit, so I'll replace `pass` with `return None`.

Reply:
ah that does help